tests/containers/docker.pm

```diff
@@ -62,26 +66,21 @@ sub run {
 
 sub post_fail_hook {
     my $self = shift;
-    cleanup();
+    cleanup($self->firewall());
     $self->SUPER::post_fail_hook;
 }
 
 sub post_run_hook {
     my $self = shift;
-    cleanup();
+    cleanup($self->firewall());
     $self->SUPER::post_run_hook;
 }
 
-sub install_and_start_firewalld() {
-    zypper_call('install firewalld');
-    systemctl('start firewalld');
-    systemctl('restart docker');
-    $stop_firewall = 1;
-}
-
 # must ensure firewalld is stopped, if it is only enabled in this test (e.g. publiccloud test runs)
 sub cleanup() {
-    script_run('systemctl stop firewalld; systemctl restart docker') if $stop_firewall;
+    my $firewall = shift;
+    disable_and_stop_service($firewall) if $stop_firewall;
```

Comment:
Why do we force stop the firewall here even on systems where firewalld has been installed by default and has been started by default? Doesn't this potentially invalidate other tests later on in the run that will assume the default installed firewall is present and running?

Reply:
But this is only executed in the case, that `firewalld` is installed, see https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/2a4f67b08e47ae96a1a5c9f6bc2ff092ab03c19b/tests/containers/docker.pm#L53

Reply:
Ahhhh, I now understand what you mean. See https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/13758#discussion_r760380702, I think the source of confusion is elsewhere.

Reply:
This should be also fixed now. Please check.